Source/Particles/Deposition/TemperatureDeposition.H

```diff
@@ -0,0 +1,463 @@
+/* Copyright 2025 The WarpX Community
+ *
+ * This file is part of WarpX.
+ *
+ * Authors: S. Eric Clark (Helion Energy)
+ *
+ * License: BSD-3-Clause-LBNL
+ */
+
+#ifndef WARPX_TEMPERATUREDEPOSITION_H_
+#define WARPX_TEMPERATUREDEPOSITION_H_
+
+#include "Particles/Deposition/VarianceAccumulationBuffer.H"
+#include "Particles/Deposition/SharedDepositionUtils.H"
+#include "ablastr/parallelization/KernelTimer.H"
+#include "Particles/Pusher/GetAndSetPosition.H"
+#include "Particles/ShapeFactors.H"
+#include "Utils/TextMsg.H"
+#include "Utils/WarpXAlgorithmSelection.H"
+#include "Utils/WarpXConst.H"
+#ifdef WARPX_DIM_RZ
+#   include "Utils/WarpX_Complex.H"
+#endif
+
+#include "WarpX.H" // todo: remove include and pass globals as args
+
+#include <AMReX.H>
+#include <AMReX_Arena.H>
+#include <AMReX_Array4.H>
+#include <AMReX_Dim3.H>
+#include <AMReX_REAL.H>
+
+namespace warpx::particles::deposition {
+
+AMREX_GPU_HOST_DEVICE AMREX_INLINE
+void varianceDepositionSubKernel(
+    amrex::ParticleReal vx,
+    amrex::ParticleReal vy,
+    amrex::ParticleReal vz,
+    const amrex::IntVectND<3>& ixv,
+    const amrex::IntVectND<3>& iyv,
+    const amrex::IntVectND<3>& izv,
+    amrex::Real wqx_var,
+    amrex::Real wqy_var,
+    amrex::Real wqz_var,
+    const amrex::Array4<amrex::Real> & varx_arr,
+    const amrex::Array4<amrex::Real> & vary_arr,
+    const amrex::Array4<amrex::Real> & varz_arr,
+    const amrex::Array4<amrex::Real> & wx_arr,
+    const amrex::Array4<amrex::Real> & wy_arr,
+    const amrex::Array4<amrex::Real> & wz_arr,
+    const amrex::Array4<amrex::Real> & w2x_arr,
+    const amrex::Array4<amrex::Real> & w2y_arr,
+    const amrex::Array4<amrex::Real> & w2z_arr,
+    const amrex::Array4<amrex::Real> & vxbar_arr,
+    const amrex::Array4<amrex::Real> & vybar_arr,
+    const amrex::Array4<amrex::Real> & vzbar_arr
+)
+{
```

Comment:
Looking at this closely, unfortunately this doesn't look thread safe. The RHS depends on the LHS, but other threads maybe contributing to the quantities so they become inconsistent. For example, between setting `wx_old` and `wxsum`, other threads may also be adding to `wx_arr` so the `old` and `sum` are not consistent. And the same for other quantities below.

Reply:
So I have been thinking about this a bit and I will ultimately use a blocking semaphore or mutex to control this over multiple operations. I need to figure out the best way to do it in the AMReX sense using atomic_ops. Likely will need a slight change in AMReX to make it work appropriately.